text/0041-shutdown.md

```diff
@@ -79,38 +79,38 @@ impl Shutdown {
         Self::default()
     }
 
-    // Registers a shutdown notification channel by providing its sender half.
-    fn add_notifier(&mut self, notifier: ShutdownNotifier) {
-        self.notifiers.push(notifier);
-    }
-
     // Registers a worker shutdown future, which completes once the worker terminates in one
-    // way or another.
-    fn add_worker_shutdown(&mut self, worker: impl Future<Output = Result<(), WorkerError>> + Unpin + 'static) {
+    // way or another, and registers a shutdown notification channel by providing its sender half.
+    fn add_worker_shutdown(
+        &mut self,
+        worker: impl Future<Output = Result<(), WorkerError>> + Unpin + 'static,
+        notifier: ShutdownNotifier,
```

Comment:
I think it would make more sense to reverse the params here. It makes sense before the notifier is created before the worker. That would also match the order in the struct and in the body of this func.

Reply:
okay 🙂